service/history/ndc/history_replicator.go

```diff
@@ -732,24 +645,20 @@ func (r *HistoryReplicatorImpl) applyNonStartEventsResetWorkflow(
 	mutableState workflow.MutableState,
 	task replicationTask,
 ) error {
-
-	requestID := uuid.New() // requestID used for start workflow execution request.  This is not on the history event.
-	stateBuilder := r.newStateBuilder(mutableState, task.getLogger())
-	_, err := stateBuilder.ApplyEvents(
-		ctx,
-		task.getNamespaceID(),
-		requestID,
-		*task.getExecution(),
-		task.getEvents(),
-		task.getNewEvents(),
-	)
+	mutableState, newMutableState, err := r.mutableStateMapper.ApplyEvents(ctx, wfContext, mutableState, task)
 	if err != nil {
 		task.getLogger().Error(
-			"nDCHistoryReplicator unable to apply events when applyNonStartEventsResetWorkflow",
+			"HistoryReplicator::applyNonStartEventsResetWorkflow unable to apply events",
 			tag.Error(err),
 		)
 		return err
 	}
+	if newMutableState != nil {
```

Comment:
Q: So we ignored this case before this change? Could be a bug?

Reply:
before we impl replication task batching, this case will not happen.